ui/src/app/pages/analysis/runtime-panel.tsx

```diff
@@ -243,55 +394,52 @@ export const RuntimePanel = fp.flow(withCurrentWorkspace(), withCdrVersions())((
     </div>
     {/* TODO(RW-5419): Cost estimates go here. */}
     <div style={styles.controlSection}>
-      {/* Recommended runtime: pick from default templates or change the image. */}
-      <PopupTrigger side='bottom'
-                    closeOnClick
-                    content={
-                      <React.Fragment>
-                        {
-                          fp.flow(
-                            fp.filter(({runtimeTemplate}) => hasMicroarrayData || !runtimeTemplate.dataprocConfig),
-                            fp.toPairs,
-                            fp.map(([i, preset]) => {
-                              return <MenuItem
-                              style={styles.presetMenuItem}
-                              key={i}
-                              aria-label={preset.displayName}
-                              onClick={() => {
-                                // renaming to avoid shadowing
-                                const {runtimeTemplate} = preset;
-                                const {presetDiskSize, presetMachineName, presetCompute} = fp.cond([
-                                  // Can't destructure due to shadowing.
-                                  [() => !!runtimeTemplate.gceConfig, (tmpl: Runtime) => ({
-                                    presetDiskSize: tmpl.gceConfig.diskSize,
-                                    presetMachineName: tmpl.gceConfig.machineType,
-                                    presetCompute: ComputeType.Standard
-                                  })],
-                                  [() => !!runtimeTemplate.dataprocConfig, ({dataprocConfig: {masterDiskSize, masterMachineType}}) => ({
-                                    presetDiskSize: masterDiskSize,
-                                    presetMachineName: masterMachineType,
-                                    presetCompute: ComputeType.Dataproc
-                                  })]
-                                ])(runtimeTemplate);
-                                const presetMachineType = fp.find(({name}) => name === presetMachineName, validLeonardoMachineTypes);
+      <CostEstimator
+          freeCreditsRemaining={creatorFreeCreditsRemaining}
+          profile={profile}
+          runningCost={machineRunningPrice({
+            computeType: selectedCompute,
```

Comment:
Lots of repeated code here with these 4 parameters, perhaps this should just be a single object payload, then let the `CostEstimator` pass these through to the utilities?